---
title: "Blast for Anna York"
date: '2022-11-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
library(readxl)
library(hrbrthemes)
library(stringr)
library(heatmaply)
source('./R/plot.func.addition.R')

library (devtools)
library (tidyverse)
#install.packages("BiocManager")
#BiocManager::install("Biostrings")
library(Biostrings)
```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger
-   Funding: Merck Investigator Studies Program

## Summary
These are insertions identified from a pangenome analysis conducted by Chrispin Chaguza. This code prepares for blast.

```{r}
a1 <- read_excel( './Data/22F_15BC_Insertions_For_Dan.xlsx') %>%
mutate(allele=paste(Gene.cluster, Gene.cluster.id, sep='_')) %>%
  dplyr::select(allele ,Sequence)




```

### prepare  for BLAST

```{r, eval=F}


source_url("https://raw.githubusercontent.com/lrjoshi/FastaTabular/master/fasta_and_tabular.R")

 a1 %>% select(allele, Sequence) %>%
  write.csv('./Data/genes.csv', row.names = F)
#TabularToFasta("./Data/genes.csv")
 
 source_url("https://raw.githubusercontent.com/lrjoshi/FastaTabular/master/fasta_and_tabular.R")


 prot1 <- a1 %>% select(allele, Sequence)

 prot2 <-  apply(prot1,1, function(x){
      as.character(translate(DNAString(x['Sequence'])))
      }, simplify=F) %>%
   unlist()

 
 prot3 <- cbind.data.frame('allele'=a1$allele, 'aa'=prot2 ) %>% 
   mutate(allele=paste0('gene_',allele)) %>%   
   write.csv('./Data/aa_york.csv', row.names = F)

TabularToFasta("./Data/aa_york.csv")  #note this generates a FASTA file on the home directory called dna_fasta.fasta

#IN tblastn, upload AA sequence, BLAST against TIGR4 specifically, then 'download all' as hit table csv. note use tblastn NOT blastn
```


