---
title: "Heatmaps of gene exchange experiment"
date: '2022-11-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
library(hrbrthemes)
library(stringr)
library(heatmaply)
source('./R/plot.func.addition.R')

```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger
-   Funding: Merck Investigator Studies Program

## Summary

These plots represent pangenomes created based on the sequenced strains. The heatmaps indicate whether a particular strain has **gained a gene (yellow)** or **lost a gene(dark blue)** compared to the reference. The plots at the bottom show the genomic location of genes gained and lost, mapped to a TIGR4 reference

```{r}
# key0 <-  read.table(file = './Data/Strain.clone.info.tsv', sep = '\t', header = TRUE) %>% 
#   dplyr::rename(parent_ST=ID, new_ST=Strain , strain=Hartwell)


#a1a <- read.csv('./Data/PathogenWatch/serotype(1).csv')
#a1b <- read.csv('./Data/PathogenWatch/strain(1).csv') %>% rename(gpsc=Lineage)


a1 <- read.table('./Data/pangenome/York.gene.presence.absence.15BC.tsv', header=T) %>%
  dplyr::select(-sample.id, -serotype.pcr,-sample.name.full,-sample.type, -sequencing.run, -sample.sequenced.clones,-serotype.genome) %>%
  reshape2::melt(.,id.vars=c('serotype.parental.strain','serotype.switched.strain','gpsc.lineage')) %>%
  rename( gene.cluster.name=variable)

a2 <- read.table('./Data/pangenome/York.gene.presence.absence.22F.tsv', header=T)%>%
  mutate(keep = if_else( serotype.switched.strain=='24F' & sample.id=='SP_307',1, if_else( serotype.switched.strain !='24F',1,0)))%>%
  filter(keep==1) %>%
  dplyr::select(-sample.id, -keep,-serotype.pcr,-sample.name.full,-sample.type, -sequencing.run, -sample.sequenced.clones,-serotype.genome) %>%
  reshape2::melt(.,id.vars=c('serotype.parental.strain','serotype.switched.strain','gpsc.lineage')) %>%
  rename( gene.cluster.name=variable)

b1 <- bind_rows( a1,a2)

# a value of "1" means inserted

c1 <- read.table('./Data/pangenome/York.gene.presence.absence.annotation.tsv', header=T) %>%
  full_join(b1, by='gene.cluster.name') %>%
  filter(value==1) %>%
  group_by(gene.cluster.annotation) %>%
  mutate(N_insert=n()) %>%
  filter(N_insert>1 & gene.cluster.annotation  !='hypothetical protein')


write.csv(c1,'./Results/insertions.csv')



ds1a <- read.table('./Data/pangenome/York.gene.presence.absence.annotation.tsv', header=T) %>%
  full_join(b1, by='gene.cluster.name') %>%
  filter(serotype.parental.strain=='15BC' ) %>%
  mutate(geneid=paste0(gene.cluster.name,gene.cluster.annotation))

ds1b <- ds1a %>% dplyr::select(-serotype.parental.strain,-gpsc.lineage, -gene.cluster.annotation,-gene.cluster.name,-gene.cluster.non.unique.name,-rep.gene.sequence) %>%
  reshape2::dcast(serotype.switched.strain~geneid, value.var='value') 

ds1c <- ds1b %>%
  dplyr::select(-serotype.switched.strain) %>%
  as.matrix() 

row.names(ds1c) <- ds1b$serotype.switched.strain

 heatmaply(ds1c, 
               #dendrogram = "row",
               xlab = "", ylab = "", 
               main = "",
               scale = "none",
               margins = c(60,100,40,20),
               grid_gap=0,
               titleX = FALSE,
               hide_colorbar = TRUE,
               branches_lwd = 0.1,
               label_names = c("Serotype/isolate", "Feature:", "Value"),
               fontsize_row = 5, fontsize_col = 5,
               labCol = colnames(ds1c),
               labRow = rownames(ds1c),
               heatmap_layers = theme(axis.line=element_blank()
                                      )
 )
```

nspC/aguA locus acquired with 8, 24F, 15BC on GPSC19;at least a couple of these seem to be in an operon (AguA, nspC):

also notable that speA exchanged on a couple of the switches (this looks like maybe an allelic exchange>)

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6727871/

