---
title: "Heatmaps of gene exchange experiment"
date: '2022-11-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
library(hrbrthemes)
library(stringr)
library(heatmaply)
source('./R/plot.func.addition.R')

```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger
-   Funding: Merck Investigator Studies Program

## Summary

These plots represent pangenomes created based on the sequenced strains. The heatmaps indicate whether a particular strain has **gained a gene (yellow)** or **lost a gene(dark blue)** compared to the reference. The plots at the bottom show the genomic location of genes gained and lost, mapped to a TIGR4 reference

```{r}
# key0 <-  read.table(file = './Data/Strain.clone.info.tsv', sep = '\t', header = TRUE) %>% 
#   dplyr::rename(parent_ST=ID, new_ST=Strain , strain=Hartwell)


#a1a <- read.csv('./Data/PathogenWatch/serotype(1).csv')
#a1b <- read.csv('./Data/PathogenWatch/strain(1).csv') %>% rename(gpsc=Lineage)


a1 <- read.table('./Data/pangenome/York.gene.presence.absence.15BC.tsv', header=T) %>%
  dplyr::select(-sample.id, -serotype.pcr,-sample.name.full,-sample.type, -sequencing.run, -sample.sequenced.clones,-serotype.genome) %>%
  reshape2::melt(.,id.vars=c('serotype.parental.strain','serotype.switched.strain','gpsc.lineage')) %>%
  rename( gene.cluster.name=variable)

a2 <- read.table('./Data/pangenome/York.gene.presence.absence.22F.tsv', header=T)%>%
  dplyr::select(-sample.id, -serotype.pcr,-sample.name.full,-sample.type, -sequencing.run, -sample.sequenced.clones,-serotype.genome) %>%
  reshape2::melt(.,id.vars=c('serotype.parental.strain','serotype.switched.strain','gpsc.lineage')) %>%
  rename( gene.cluster.name=variable)

b1 <- bind_rows( a1,a2)

# a value of "0" means the gene is present/absent in both the ref WT and capsule-switched strain, "1" means present in the WT but not the capsule-switched strain (i.e., possibly deleted), and "2" means absence in the WT but present in the capsule-switched strain (i.e., possibly inserted).

c1 <- read.table('./Data/pangenome/York.gene.presence.absence.annotation.tsv', header=T) %>%
  left_join(b1, by='gene.cluster.name') %>%
  filter(value==2) %>%
  group_by(gene.cluster.annotation) %>%
  mutate(N_insert=n()) %>%
  filter(N_insert>1 & gene.cluster.annotation  !='hypothetical protein')






```

nspC/aguA locus acquired with 8, 24F, 15BC on GPSC19;at least a couple of these seem to be in an operon (AguA, nspC):

also notable that speA exchanged on a couple of the switches (this looks like maybe an allelic exchange>)

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6727871/

