---
title: "Heatmaps of gene exchange experiment"
date: '2022-11-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
library(hrbrthemes)
library(heatmaply)
source('./R/plot.func.addition.R')

```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger

```{r}
colorize <- function(x, color) {
  if (knitr::is_latex_output()) {
    sprintf("\\textcolor{%s}{%s}", color, x)
  } else if (knitr::is_html_output()) {
    sprintf("<span style='color: %s;'>%s</span>", color,
      x)
  } else x
}
```

## Summary

These plots represent pangenomes created based on the sequenced strains. The heatmaps indicate whether a particular strain has `r colorize("gained a gene", "#FDE725FF")` or `r colorize("lost a gene", "#404788FF")` compared to the reference

```{r}
key1 <-  read.table(file = './Data/Strain.clone.info.tsv', sep = '\t', header = TRUE) %>% rename(parent_ST=ID, new_ST=Strain , strain=Hartwell)

g1 <- read.table(file = './Data/Group1 gene.presence_absence.tsv', sep = '\t', header = TRUE)

g2 <- read.table(file = './Data/Group2 gene.presence_absence.tsv', sep = '\t', header = TRUE)

g3 <- read.table(file = './Data/Group3 gene.presence_absence.tsv', sep = '\t', header = TRUE)

a1 <- read.table(file = './Data/Genes.with.variable.presence.patterns_with_annotations.tsv', sep = '\t', header = TRUE) %>%
  mutate(Gene= gsub('~~~~~~','......', fixed=T, Gene ),Non.unique.Gene.name = gsub(';','',Non.unique.Gene.name,fixed=T)) %>%
  filter(Non.unique.Gene.name!='') %>%
  arrange(Non.unique.Gene.name) %>%
  group_by(Non.unique.Gene.name) %>%
  mutate(allele = paste0(Non.unique.Gene.name,'_', row_number()),
        ) %>%
  ungroup() %>%
  select(allele,Cps.Gene, Gene,Non.unique.Gene.name, Sequence)

```

## Serotype 9V background

Parent strain is Serotype 9V ID ABC010001831, obtained from the CDC

```{r, fig.width=10, fig.height=5}
p1 <- plot.func.addition(g1)

p1$p
```

## Serotype 15B/C background

Parent strain is 15B/C, a carriage isolate from Hungary, ID Pool_hung_H10 (H124)

```{r, fig.width=10, fig.height=5}
p2 <- plot.func.addition(g2)
p2$p
```

## Serotype 22F background

Parent strain is serotype 22F, ID: ABC010003966, obtained from the CDC (IPD isolate)

```{r, fig.width=10, fig.height=5}
p3 <- plot.func.addition(g3)
p3$p
```

```{r}
p1$gain_loss <- p1$gain_loss %>%
  mutate(parentST='9V')

p2$gain_loss <- p2$gain_loss %>%
  mutate(parentST='15BC')

p3$gain_loss <- p3$gain_loss %>%
  mutate(parentST='22F')
  

#here we are looking for strains that have gain of one allele and loss of another..see 15B.C #3; gpsB_1
combined <- bind_rows(p1$gain_loss, p2$gain_loss, p3$gain_loss) %>%
  filter(value %in% c(-1,1)) %>%
  arrange(Var1, Var2)
```


```{r, eval=F}
#prepare  for BLAST

library (devtools)
library (tidyverse)
source_url("https://raw.githubusercontent.com/lrjoshi/FastaTabular/master/fasta_and_tabular.R")

 a1 %>% select(allele, Sequence) %>%
  write.csv('./Data/genes.csv', row.names = F)
TabularToFasta("./Data/genes.csv")

#IN BLAST, upload, BLAST against TIGR4 specifically, then 'download all' as hit table csv
```

## Genomic context of gain/loss based on mapping to TIGR4 genome

Set 1
```{r}
loc1 <- read.csv('./Data/tigr4-blast-Alignment-HitTable.csv', header=F) %>%
  filter(V2=='CP089948.1') %>%
  dplyr::rename(allele=V1, start=V9, end=V10) %>%
  select(allele, start, end)

p1a <- p1$gain_loss %>%
  dplyr::rename(strain=Var1, allele=Var2) %>%
  left_join(loc1, by='allele') %>%
  filter(value %in% c(-1,1) & !is.na(start)) %>%
  mutate(value=as.factor(value) ,strainN=as.numeric(as.factor(strain)))

p2a <- ggplot(p1a, aes( y=strainN, x=start ,color=value,text=paste(strain,allele), alpha=0.5 )) +
         geom_point(size=0.3) +
       theme_classic() +
  geom_vline(xintercept=c(315597, 339358)) #roughly dexB and aliA

ggplotly(p2a)
```
Set 2

```{r}

p1a <- p2$gain_loss %>%
  dplyr::rename(strain=Var1, allele=Var2) %>%
  left_join(loc1, by='allele') %>%
  filter(value %in% c(-1,1) & !is.na(start)) %>%
  mutate(value=as.factor(value) ,strainN=as.numeric(as.factor(strain)))

p2a <- ggplot(p1a, aes( y=strainN, x=start ,color=value,text=paste(strain,allele), alpha=0.5 )) +
         geom_point(size=0.3) +
       theme_classic() +
  geom_vline(xintercept=c(315597, 339358)) #roughly dexB and aliA

ggplotly(p2a)
```
 Set 3
 

```{r}

p1a <- p3$gain_loss %>%
  dplyr::rename(strain=Var1, allele=Var2) %>%
  left_join(loc1, by='allele') %>%
  filter(value %in% c(-1,1) & !is.na(start)) %>%
  mutate(value=as.factor(value) ,strainN=as.numeric(as.factor(strain)))

p2a <- ggplot(p1a, aes( y=strainN, x=start ,color=value,text=paste(strain,allele), alpha=0.5 )) +
         geom_point(size=0.3) +
       theme_classic() +
  geom_vline(xintercept=c(315597, 339358)) #roughly dexB and aliA

ggplotly(p2a)
```




