---
title: "Heatmaps of gene exchange experiment"
date: '2022-11-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
library(hrbrthemes)
library(heatmaply)
source('./R/plot.func.R')
source('./R/plot.func.addition.R')

```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger

```{r}
colorize <- function(x, color) {
  if (knitr::is_latex_output()) {
    sprintf("\\textcolor{%s}{%s}", color, x)
  } else if (knitr::is_html_output()) {
    sprintf("<span style='color: %s;'>%s</span>", color,
      x)
  } else x
}
```

## Summary

These plots represent pangenomes created based on the sequenced strains. The heatmaps indicate whether a particular strain has `r colorize("gained a gene", "#FDE725FF")` or `r colorize("lost a gene", "#404788FF")` compared to the reference

```{r}
key1 <-  read.table(file = './Data/Strain.clone.info.tsv', sep = '\t', header = TRUE) %>% rename(parent_ST=ID, new_ST=Strain , strain=Hartwell)

g1 <- read.table(file = './Data/Group1 gene.presence_absence.tsv', sep = '\t', header = TRUE)

g2 <- read.table(file = './Data/Group2 gene.presence_absence.tsv', sep = '\t', header = TRUE)

g3 <- read.table(file = './Data/Group3 gene.presence_absence.tsv', sep = '\t', header = TRUE)

a1 <- read.table(file = './Data/Genes.with.variable.presence.patterns_with_annotations.tsv', sep = '\t', header = TRUE) %>%
  mutate(Gene= gsub('~~~~~~','......', fixed=T, Gene ),Non.unique.Gene.name = gsub(';','',Non.unique.Gene.name,fixed=T)) %>%
  filter(Non.unique.Gene.name!='') %>%
  arrange(Non.unique.Gene.name) %>%
  group_by(Non.unique.Gene.name) %>%
  mutate(allele = paste0(Non.unique.Gene.name,'_', row_number()),
        ) %>%
  ungroup() %>%
  select(allele,Cps.Gene, Gene,Non.unique.Gene.name)

```

## Serotype 9V background

Parent strain is Serotype 9V ID ABC010001831, obtained from the CDC

```{r, fig.width=10, fig.height=5}
plot.func.addition(g1)

```

## Serotype 15B/C background

Parent strain is 15B/C, a carriage isolate from Hungary, ID Pool_hung_H10 (H124)

```{r, fig.width=10, fig.height=5}
plot.func.addition(g2)

```

## Serotype 22F background

Parent strain is serotype 22F, ID: ABC010003966, obtained from the CDC (IPD isolate)

```{r, fig.width=10, fig.height=5}
plot.func.addition(g3)

```
