---
title: "Untitled"
author: "Dan Weinberger"
date: '2023-08-15'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(plotly)
#library(d3heatmap)
library(viridis)
plots <- readRDS('./Data/plots.rds')
```

## Credits

-   Analysis by Chrispin Chaguza
-   Experiments by Anna York
-   Sequencing by Jason Rosch and colleagues (St Jude's)
-   Interactive plots by Dan Weinberger
-   Funding: Merck Investigator Studies Program

## Summary

These plots represent pangenomes created based on the sequenced strains. The heatmaps indicate whether a particular strain has **gained a gene (yellow)** or **lost a gene(dark blue)** compared to the reference. The plots at the bottom show the genomic location of genes gained and lost, mapped to a TIGR4 reference

## Presence absence for 15BC

```{r}
plots$presence_15bc$p
```

## Presence absence for 22F

```{r}
plots$presence22f$p
```


## Map of presence absence  for 15BC variants to TIGR4 genome

```{r}
ggplotly(plots$map15bc)
```


## Map of presence absence for 22F variants to TIGR4 genome

```{r}
ggplotly(plots$map22f)
```

